### PR TITLE
Communities: create search community records endpoint

### PR DESCRIPTION
**Pull request by Pineirin**
Define tests for the created endpoint.
closes: inveniosoftware/invenio-communities#408

### CODE REVIEW

File: invenio_rdm_records/records/mappings/v7/rdmrecords/records/record-v4.0.0.json [hunk 1]
```diff
@@ -37,6 +37,13 @@
           "id" : {
             "type" : "keyword"
           },
+          "communities": {
+            "properties": {
+              "default": {
+                "type": "keyword"
```
Comment (ntarocco) [inline invenio_rdm_records/records/mappings/v7/rdmrecords/records/record-v4.0.0.json:43]: can you please add the `ids` field too? 
Did you check if there is any other field missing?

File: invenio_rdm_records/resources/resources.py [hunk 2]
```diff
@@ -43,6 +44,22 @@ def p(route):
     #
     # Review request
     #
+    @request_search_args
+    @request_view_args
+    @response_handler(many=True)
+    def search_community_records(self):
```
Comment (ntarocco) [inline invenio_rdm_records/resources/resources.py:50]: Can you please move this down at the very bottom with a similar comment on top?
the comment here says `# Review request` and it is not releated to that

File: tests/services/test_rdm_service.py [hunk 3]
```diff
@@ -132,3 +180,87 @@ def test_embargo_lift_with_error(running_app, es_clear, minimal_record):
...
+    )
+    draft = service.create(superuser_identity, minimal_record)
+    record = service.publish(id_=draft.id, identity=superuser_identity)
+
+    # Record should not be lifted since it didn't expire (until 3220)
+    with pytest.raises(EmbargoNotLiftedError):
+        service.lift_embargo(_id=record['id'], identity=superuser_identity)
+
+
+def prueba(app, location, db, es_clear, vocabularies, users):
```
Comment (ntarocco) [inline tests/services/test_rdm_service.py:219]: ?

File: tests/services/test_rdm_service.py [hunk 4]
```diff
@@ -132,3 +180,87 @@ def test_embargo_lift_with_error(running_app, es_clear, minimal_record):
     # Record should not be lifted since it didn't expire (until 3220)
     with pytest.raises(EmbargoNotLiftedError):
         service.lift_embargo(_id=record['id'], identity=superuser_identity)
+
+
+def test_embargo_lift_with_error(running_app, es_clear, minimal_record):
```
Comment (ntarocco) [inline tests/services/test_rdm_service.py:185]: did you add these tests?

File: invenio_rdm_records/resources/resources.py (hunk 2 above)
Comment (ntarocco) [inline invenio_rdm_records/resources/resources.py:50]: It does not look like that you have created a route for this endpoint, and there is no test for this resource. The endpoint cannot be used.
The URL, I think, should be `/communities/<pid_value>/records`

File: invenio_rdm_records/services/services.py [hunk 5]
```diff
@@ -86,3 +88,38 @@ def scan_expired_embargos(self, identity):
             f"AND access.embargo.until:[* TO {today}]"
 
         return self.scan(identity=identity, q=embargoed_q)
+
+    def search_community_records(self, identity, community, params=None,
```
Comment (ntarocco) [inline invenio_rdm_records/services/services.py:92]: This signature does not match with how you are using it is the resource.

File: invenio_rdm_records/services/services.py [hunk 6]
```diff
@@ -86,3 +88,38 @@ def scan_expired_embargos(self, identity):
...
+                                 es_preference=None, **kwargs):
+        """Search for drafts records matching the querystring."""
+        community_uuid = community.to_dict()['uuid']
```
Comment (ntarocco) [inline invenio_rdm_records/services/services.py:95]: You are already getting the `id` from the resource, not the community.

File: invenio_rdm_records/services/services.py [hunk 7]
```diff
@@ -86,3 +88,38 @@ def scan_expired_embargos(self, identity):
...
+
+        search_result = self._search(
+            "search",
+            identity,
+            params,
+            es_preference,
+            record_cls=self.record_cls,
+            search_opts=self.config.search,
+            extra_filter=Q(
+                "term", **{"parent.communities.default": str(community_uuid)}
```
Comment (ntarocco) [inline invenio_rdm_records/services/services.py:109]: I am not sure this is enough. A record can be in the default community, or in one of the `ids` list.
Please add tests for both cases.

**Review by ntarocco — COMMENTED** [review]
Much better, good job! Added some more comments because I think that there is still something to add.

**Review by Pineirin — COMMENTED** [review]
Why was this appearing as an error to you but not to me?

**Review by zzacharo — APPROVED** [review]

**Pull request merged**